akka-projection-core/src/main/scala/akka/projection/Projection.scala

```diff
@@ -7,11 +7,23 @@ package akka.projection
 import akka.Done
 import akka.actor.ClassicActorSystemProvider
 import akka.annotation.ApiMayChange
+import akka.stream.scaladsl.Source
 
-import scala.concurrent.{ ExecutionContext, Future }
+import scala.concurrent.Future
 
 @ApiMayChange
 trait Projection {
-  def start()(implicit systemProvider: ClassicActorSystemProvider): Unit
+
+  def run()(implicit systemProvider: ClassicActorSystemProvider): RunningProjection
+
+  /**
+   * This method should return the projection Source with the user event handler applied to it, but before any sink attached.
+   */
+  private[projection] def mappedSource: Source[Done, _]
```

Comment:
In order to support TestSink in tests, we need to have a handle to the Source before it gets materialized, but after the user event handler is attached to it.

Reply:
This increases a little bit the complexity of the `Projection` trait, but we don't expect it to be an widely used trait. Only Projection implementors are exposed to it.

Reply:
We can make it `protected` and tag it with `@InternalStableApi`.

Reply:
If we want to allow end users to build their own projection implementations then it should probably be available to them (as `protected`?)

Reply:
Actually `protected` won't work because the TestKit needs to be able to access it. It needs to me public API.